crates/tags/src/tag.rs

```diff
@@ -1,60 +1,240 @@
 // SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
 // SPDX-License-Identifier: Apache-2.0
 
-use std::hash::Hash;
 use std::ops::{Deref, DerefMut};
 use std::str::FromStr;
 
-use drawbridge_http::http::{self, Error, Request, StatusCode};
+use drawbridge_http::http::Request;
 use drawbridge_http::{async_trait, FromRequest};
+use drawbridge_jose::jws::Jws;
+use drawbridge_type::Entry;
 
-use semver::{self, Version};
+use http_types;
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub struct Tag(Version);
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct TagName(String);
 
-impl Deref for Tag {
-    type Target = Version;
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TagValue {
+    pub body: Vec<u8>,
+    pub kind: Kind,
+    pub name: TagName,
+}
+
+impl TagValue {
+    #[allow(dead_code)]
+    pub const TYPE_ENTRY: &'static str = "application/vnd.drawbridge.tag.v1+json";
+    #[allow(dead_code)]
+    pub const TYPE_JOSE: &'static str = "application/jose+json";
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+// value of the HashMap, name is the Key
```

Comment:
This comment should be either above the attributes or below, but not in the middle